src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,190 @@
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.AbsoluteEncoderConfig;
+import com.revrobotics.spark.config.ClosedLoopConfig;
+import com.revrobotics.spark.config.ClosedLoopConfig.FeedbackSensor;
+import com.revrobotics.spark.config.SparkMaxConfig;
+import edu.wpi.first.units.measure.Distance;
+import edu.wpi.first.units.measure.LinearVelocity;
+import edu.wpi.first.units.measure.Time;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.shuffleboard.ComplexWidget;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+
+public class Elevator extends SubsystemBase {
+  SparkMax elevatorMotor;
+  SparkAbsoluteEncoder elevatorEncoder;
+  private SparkMaxSim simElevatorMotor;
+  private final Time sparkPeriod;
+  private static Mechanism2d mech = new Mechanism2d(10, 10);
+  private static ComplexWidget widg;
+  private MechanismLigament2d elevatorMech;
+
+  public static final AbsoluteEncoderConfig encoderCfg =
+      new AbsoluteEncoderConfig()
+          .positionConversionFactor(Constants.ElevatorConstants.elevatorConversion);
+  public static final ClosedLoopConfig loopCfg =
+      new ClosedLoopConfig()
+          .feedbackSensor(FeedbackSensor.kPrimaryEncoder)
+          .pidf(
+              Constants.ElevatorConstants.elevatorP,
+              Constants.ElevatorConstants.elevatorI,
+              Constants.ElevatorConstants.elevatorD,
+              Constants.ElevatorConstants.elevatorFF);
+  private final ElevatorSim m_elevatorSim;
+
+  public Elevator() {
+    sparkPeriod = Millisecond.one();
+    elevatorMotor = new SparkMax(1, MotorType.kBrushless);
+    elevatorEncoder = elevatorMotor.getAbsoluteEncoder();
+    m_elevatorSim =
+        new ElevatorSim(
+            m_elevatorGearbox,
+            kElevatorGearing,
+            kCarriageMass.in(Kilograms),
+            kElevatorDrumRadius.in(Meters),
+            kMinElevatorHeight.in(Meters),
+            kMaxElevatorHeight.in(Meters),
+            true,
+            0);
+    elevatorMotor.configure(
+        new SparkMaxConfig().apply(encoderCfg).apply(loopCfg),
+        ResetMode.kNoResetSafeParameters,
+        PersistMode.kNoPersistParameters);
+    if (Robot.isSimulation()) {
+      simElevatorMotor =
+          new SparkMaxSim(elevatorMotor, Constants.ElevatorSimConstants.m_elevatorGearbox);
+      elevatorMech =
+          mech.getRoot("root", 5, 5)
```

Comment:
This is were you would reduce this size. You may also need to reduce the +1 hight you have on the mech